<launch>
<!-- 
	run robot alongside with:
	roslaunch nicol_controller NICOL_launch_all.launch use_platform:=false launch_tf_publisher:=true use_moveit:=false use_primitives:=false use_face_expr:=false use_cameras:=false with_left_omp:=false compat_mode:=false use_speech_recognition:=false
 -->

	<!-- marker cam params -->
	<arg name="use_markers_camera"  default="true" />
	<arg name="serial_no"           default="822512060411"/>
	<arg name="enable_depth"        default="false"/>
	<arg name="depth_width"         default="1280"/>
	<arg name="depth_height"        default="720"/>  
	<arg name="depth_fps"           default="30"/>
	<arg name="enable_color"        default="true"/>
	<arg name="color_width"         default="1920"/>
	<arg name="color_height"        default="1080"/>
	<arg name="color_fps"           default="6"/>
	<arg name="filters"                   default="colorizer"/>
	<arg name="clip_distance"             default="-2"/>
	<arg name="align_depth"               default="true"/>

	<arg name="play_bag"               default="false"/>

	<!-- <arg name="use_wrist_camera" default="false" />
	<arg name="use_top_camera" default="false" /> -->

	<!-- model args -->
	<arg name="theta"      default="0" doc="Angle of wrist part wrt. z axis of link6"/>
    <arg name="x_dist"      default="-0.3527" doc="Distance along x axis"/>
	<arg name="y_dist"      default="0.01245" doc="Distance along y axis"/>
    <arg name="z_dist"      default="0.2406" doc="Distance along z axis"/>
    <arg name="attached"      default="false"/>
	<arg name="launch_rviz"      default="false"/>

	<include file="$(find nicol_rh8d)/launch/marker_cam_model.launch">
		<arg name="theta"      value="$(arg theta)"/>
		<arg name="x_dist"      value="$(arg x_dist)" />
		<arg name="y_dist"      value="$(arg y_dist)" />
		<arg name="z_dist"      value="$(arg z_dist)" />
		<arg name="attached" value="$(arg attached)"/>
		<arg name="launch_rviz" value="$(arg launch_rviz)"/>
	</include>

	<!-- common manager -->
	<node  if="$(arg use_markers_camera)" pkg="nodelet" type="nodelet" name="rs_manager" args="manager" output="screen" required="true" />
	<!-- markers cam -->
	<include if="$(eval use_markers_camera or bool(play_bag))" file="$(find nicol_rh8d)/launch/realsense.launch">
		<arg name="external_manager"    value="true"/>
		<arg name="manager"             value="/rs_manager"/>
		<arg name="use_realsense" value="$(arg use_markers_camera)" />
		<arg name="camera"             value="marker_realsense" />
		<arg name="serial_no"           value="$(arg serial_no)"/>
		<arg name="enable_depth"        value="$(arg enable_depth)"/>
		<arg name="depth_width"         value="$(arg depth_width)"/>
		<arg name="depth_height"        value="$(arg depth_height)"/>  
		<arg name="depth_fps"           value="$(arg depth_fps)"/>
		<arg name="enable_color"        value="$(arg enable_color)"/>
		<arg name="color_width"         value="$(arg color_width)"/>
		<arg name="color_height"        value="$(arg color_height)"/>
		<arg name="color_fps"           value="$(arg color_fps)"/>
		<arg name="filters"                   value="$(arg filters)"/>
		<arg name="clip_distance"             value="$(arg clip_distance)"/>
		<arg name="align_depth"               value="$(arg align_depth)"/>
	</include>

	<!-- dataset collector script -->
	<node pkg="nicol_rh8d" name="dataset_collector" type="dataset_collector.py" output="screen" required="true">
		<!-- <param name="move_hand"        type="bool"   value="$(arg move_hand)" /> -->
	</node>

</launch>
